packages/host/tests/integration/components/card-basics-test.gts

```diff
@@ -1111,75 +1111,65 @@ module('Integration | card-basics', function (hooks) {
         workExperiences: [],
       });
 
-      let changeEvent:
-        | { instance: BaseDef; fieldName: string; value: any }
-        | undefined;
-      let eventCount = 0;
+      let events: any[] = [];
       let subscriber = (instance: BaseDef, fieldName: string, value: any) => {
-        eventCount++;
-        changeEvent = {
+        events.push({
           instance,
           fieldName,
           value,
-        };
+        });
       };
       subscribeToChanges(mango, subscriber);
 
       try {
         let firstWorkExperience = new WorkExperience();
         mango.workExperiences.push(firstWorkExperience);
         await flushLogs();
-        await waitUntil(() => eventCount === 1);
-        assert.deepEqual(
-          changeEvent?.instance,
-          mango,
-          'the instance was correctly specified in change event',
-        );
-        assert.strictEqual(
-          changeEvent?.fieldName,
-          'workExperiences',
-          'the fieldName was correctly specified in change event',
-        );
         assert.deepEqual(
-          changeEvent?.value[0],
-          firstWorkExperience,
-          'the field value was correctly specified in change event',
+          events,
+          [
+            {
+              instance: mango,
+              fieldName: 'workExperiences',
+              value: [firstWorkExperience],
+            },
+          ],
+          'the change event is correct',
         );
+        events = [];
 
         firstWorkExperience.company = 'First Company';
-        await waitUntil(() => eventCount === 2);
-        assert.deepEqual(
-          changeEvent?.instance,
-          firstWorkExperience,
-          'the instance was correctly specified in change event',
-        );
-        assert.strictEqual(
-          changeEvent?.fieldName,
-          'company',
-          'the fieldName was correctly specified in change event',
-        );
+        await flushLogs();
         assert.deepEqual(
-          changeEvent?.value,
-          'First Company',
-          'the field value was correctly specified in change event',
+          events,
+          [
+            {
+              instance: firstWorkExperience,
+              fieldName: 'company',
+              value: 'First Company',
+            },
+            {
+              instance: mango,
+              fieldName: 'workExperiences.0.company',
+              value: 'First Company',
+            },
```

Comment:
this is the key event that was missing: when the interior of a compound field changes we need to notify on the enclosing card instance.